### PR TITLE
fix test suite

### PR DESCRIPTION
**Pull request by doudou**
The test suite had a lot of common boilerplate that made it hard to read. The refactoring of the
test suite itself is mainly factoring out of common patterns.

Moreover,
- the `state.received_messages` was used in a non-thread safe way
- the `has_error`/`state` fields were basically duplicating the built-in thread state tracking. Leverage the latter to remove the former - and this way get exceptions reported for better debugging.

### CODE REVIEW

File: tasks/Task.cpp [hunk 1]
```diff
@@ -110,13 +119,9 @@ bool Task::handleIncomingWebsocketMessage(char const* data, WebSocket *connectio
     if (is_controlling) {
         return handleControlMessage();
     }
-
-    if (!handleAskControlMessage()){
-        connection->close();
-        return false;
+    else {
+        return handleAskControlMessage();
```
Comment (henriquejsfj) [inline tasks/Task.cpp:123]: Why did you change your mind about closing the connection?

Also, don't need to use else.

**Review by henriquejsfj — APPROVED** [review]

File: tasks/Task.cpp (hunk 1 above)
Comment (doudou) [reply tasks/Task.cpp:123]: See the message from https://github.com/rock-drivers/drivers-orogen-controldev_websocket/pull/2/commits/aca4d7b23b55f4a4c840c2fdf8a2a886e9ae7f7b

Comment (doudou) [reply tasks/Task.cpp:123]: The if / else in this very particular instance ... is a matter of taste. 

In this case, I like it because it shows the symmetry. I would not have commented on it before, because it *really* is a matter of taste.
